src/app/games/yinsh/YinshBoard.ts

```diff
@@ -0,0 +1,47 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { HexaBoard } from 'src/app/jscaip/HexaBoard';
+import { Player } from 'src/app/jscaip/Player';
+import { ArrayUtils, NumberTable, Table } from 'src/app/utils/ArrayUtils';
+import { YinshPiece } from './YinshPiece';
+
+export class YinshBoard extends HexaBoard<YinshPiece> {
+    public static SIZE: number = 11;
+    public static EXCLUDED_CASES: ReadonlyArray<number> = [6, 4, 3, 2, 1];
+
+    public static EMPTY: YinshBoard =
+        new YinshBoard(ArrayUtils.createBiArray(YinshBoard.SIZE, YinshBoard.SIZE, YinshPiece.EMPTY));
+
+    public static of(table: Table<YinshPiece>): YinshBoard {
+        return new YinshBoard(table);
+    }
+
+    public constructor(public readonly contents: Table<YinshPiece>) {
+        super(contents, YinshBoard.SIZE, YinshBoard.SIZE, YinshBoard.EXCLUDED_CASES, YinshPiece.EMPTY);
+    }
+    public toNumberTable(): NumberTable {
+        return ArrayUtils.mapBiArray(this.contents, YinshPiece.encoder.encodeNumber);
+    }
+    public isOnBoard(coord: Coord): boolean {
+        if (HexaBoard.isOnBoard(coord, YinshBoard.SIZE, YinshBoard.SIZE, YinshBoard.EXCLUDED_CASES) === false) {
+            return false;
+        } else if (coord.x === YinshBoard.SIZE-1 && coord.y === 0) {
+            return false;
+        } else if (coord.x === 0 && coord.y === YinshBoard.SIZE-1) {
+            return false;
+        } else if (coord.x === 0 && coord.y === (YinshBoard.SIZE-1)/2) {
+            return false;
+        } else if (coord.x === YinshBoard.SIZE-1 && coord.y === (YinshBoard.SIZE-1)/2) {
+            return false;
+        }
+        return true;
```

Comment:
un isInRange + is not PieceYinsh.NONE marcherais pas plus clairement ?

Reply:
ah dans ce cas ci je vois la possible utilité de FourPieceState, c'est vraiment une approche statique vs. dynamique en fait. Statique = tu t'arranges pour jamais atteindre un mauvais état, donc on s'en fou ce qui se trouve dans le board à cet endroit; dynamique = peut-être tu liras les mauvais états, donc tu te protège contre ça avec des valeurs spéciales.
Quid d'un board qui contient des null plutôt que des pieces pour ces "mauvais" états ? Afin de pas polluer l'implémentation de la pièce ?

Reply:
Bah, je dirais que c'est mieux de pas abuser de l'usage du null, mais qu'au pire par lisibilité avoir YinshPiece.NONE = null; comme ça on utilise un nom parlant mais je préfère quand même d'éviter de nuller (pour les console.table déjà c't'un peu mieux :p ? )